src/unk_0200ACF0.c

```diff
@@ -0,0 +1,223 @@
+
+    UnkStruct_02021758 sp00;
+    sp00.charData = GF2DGfxResObj_GetCharDataPtr(obj);
+    sp00.dest = GF2DGfxResObj_GetLoadAddress(obj);
+    sp00.id = GF2DGfxResObj_GetResID(obj);
+    sp00.unk_0C = TRUE;
+    return sub_02021758(&sp00);
+}
+
+void sub_0200ADE4(GF_2DGfxResObjList *charResObjList) {
+    GF_ASSERT(charResObjList != NULL);
+    for (int i = 0; i < charResObjList->max; ++i) {
+        if (charResObjList->obj[i] != NULL) {
+            GF_ASSERT(sub_0200ADA4(charResObjList->obj[i]));
+        }
+    }
+}
+
+BOOL sub_0200AE18(GF_2DGfxResObj *obj) {
+    GF_ASSERT(obj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(obj) == GF_GFX_RES_TYPE_CHAR);
+
+    UnkStruct_02021758 sp00;
+    sp00.charData = GF2DGfxResObj_GetCharDataPtr(obj);
+    sp00.dest = GF2DGfxResObj_GetLoadAddress(obj);
+    sp00.id = GF2DGfxResObj_GetResID(obj);
+    sp00.unk_0C = TRUE;
+    return sub_020217B0(&sp00);
+}
+
+void sub_0200AE58(GF_2DGfxResObjList *charResObjList) {
+    GF_ASSERT(charResObjList != NULL);
+    for (int i = 0; i < charResObjList->max; ++i) {
+        if (charResObjList->obj[i] != NULL) {
+            GF_ASSERT(sub_0200AE18(charResObjList->obj[i]));
+        }
+    }
+}
+
+void sub_0200AE8C(GF_2DGfxResObj *obj) {
+    GF_ASSERT(obj != NULL);
+
+    int id = GF2DGfxResObj_GetResID(obj);
+    NNSG2dCharacterData *charData = GF2DGfxResObj_GetCharDataPtr(obj);
+    sub_0202183C(id, charData);
+}
+
+void sub_0200AEB0(GF_2DGfxResObj *obj) {
+    GF_ASSERT(obj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(obj) == GF_GFX_RES_TYPE_CHAR);
+
+    int id = GF2DGfxResObj_GetResID(obj);
+    sub_02021884(id);
+}
+
+void sub_0200AED4(GF_2DGfxResObjList *charResObjList) {
+    GF_ASSERT(charResObjList != NULL);
+    for (int i = 0; i < charResObjList->max; ++i) {
+        if (charResObjList->obj[i] != NULL) {
+            sub_0200AEB0(charResObjList->obj[i]);
+        }
+    }
+}
+
+NNSG2dImageProxy *sub_0200AF00(GF_2DGfxResObj *obj) {
+    GF_ASSERT(obj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(obj) == GF_GFX_RES_TYPE_CHAR);
+
+    int id = GF2DGfxResObj_GetResID(obj);
+    return sub_02021910(id);
+}
+
+NNSG2dImageProxy *sub_0200AF24(GF_2DGfxResObj *charResObj, GF_2DGfxResObj *cellResObj) {
+    NNSG2dImageProxy *ret;
+
+    GF_ASSERT(charResObj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(charResObj) == GF_GFX_RES_TYPE_CHAR);
+    GF_ASSERT(cellResObj != NULL);
+    GF_ASSERT(GF2DGfxResObj_GetResType(cellResObj) == GF_GFX_RES_TYPE_CELL);
+
+    int resID = GF2DGfxResObj_GetResID(charResObj);
+    NNSG2dCellDataBank *cellData = GF2DGfxResObj_GetCellDataPtr(cellResObj);
+    ret = sub_02021934(resID, cellData->pVramTransferData->szByteMax);
+    if (ret == NULL) {
+        ret = sub_020219B8(sub_02021910(resID));
+    }
+    return ret;
+}
+
+void sub_0200AF80(const NNSG2dImageProxy* imgProxy) {
+    GF_ASSERT(imgProxy != NULL);
+    sub_02021A50(imgProxy);
+}
+
+// ------------------
```

Comment:
?

Reply:
probably marking a border between two logical (to me at the time) groups of functions without a good-enough explanation of what that logical difference was